components/brave_savings/browser/perf_predictor_page_metrics_observer.cc

```diff
@@ -0,0 +1,92 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_savings/browser/perf_predictor_page_metrics_observer.h"
+
+#include "brave/browser/brave_browser_process_impl.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/browser/performance_monitor/system_monitor.h"
+#include "chrome/browser/sessions/session_tab_helper.h"
+#include "content/public/browser/browser_context.h"
+#include "content/public/browser/navigation_handle.h"
+#include "content/public/browser/web_contents.h"
+#include "content/public/common/content_features.h"
+#include "components/sessions/core/session_id.h"
+#include "url/gurl.h"
+#include "extensions/browser/event_router.h"
```

Comment:
I don't see where this is used?

Reply:
it also looks like you're missing a lot of includes that you should have, but they are working by "luck" because they are includes of other includes. You should always include what you use so can you please double-check all of these includes to remove any that aren't being used and add missing includes? Also make sure you add deps here https://github.com/brave/brave-core/pull/4229/files#diff-3c0e6b72a2c26894b7ad5cbd714a1a3bR16 for anything you include

Reply:
please check ^^ for all new files